code/datums/elements/blocks_explosives.dm

```diff
@@ -20,6 +20,13 @@
 
 /datum/element/blocks_explosives/Detach(datum/source)
 	. = ..()
+	if(ismovable(source))
+		var/atom/movable/target = source
+		if(length(target.locs) > 1)
+			for(var/atom/location as anything in target.locs)
+				unblock_loc(location, target.explosion_block)
+		else if(target.loc)
+			unblock_loc(target.loc, target.explosion_block)
```

Comment:
won't work, detach is called when the element is explicitly removed.
this is still good to have, but to make this work with the issues you're looking at you'll need `ELEMENT_DETACH_ON_HOST_DESTROY` as an element flag